src/diffusers/models/autoencoders/autoencoder_asym_kl.py

```diff
@@ -176,7 +176,7 @@ def forward(
             z = posterior.sample(generator=generator)
         else:
             z = posterior.mode()
-        dec = self.decode(z, sample, mask).sample
+        dec = self.decode(z, generator, sample, mask).sample
```

Comment:
Tend to agree with @tolgacangoz's comments here. It's not used in the `decode()` function. Similar to AutoencoderKL. It's used in the `forward()`:

https://github.com/huggingface/diffusers/blob/413604405fddb4692a8e9a9a9fb6c353d22881ea/src/diffusers/models/autoencoders/autoencoder_kl.py#L445

Reply:
I don't think there's a stochastic component in the `decode()` function in the first place. So, I further think there's no need to have `generator` in here too:

https://github.com/huggingface/diffusers/blob/413604405fddb4692a8e9a9a9fb6c353d22881ea/src/diffusers/models/autoencoders/autoencoder_kl.py#L286

@yiyixuxu WDYT here?

Reply:
i think this comment answered the question:
https://github.com/huggingface/diffusers/pull/5694/files#r1387349283

Reply:
@sayakpaul I had the same questions, but their explanation makes sense I think